jvm/src/main/java/com/microsoft/bond/protocol/FastBinaryReader.java

```diff
@@ -0,0 +1,147 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+
+import java.io.InputStream;
+
+public class FastBinaryReader<S extends InputStream> implements TaggedProtocolReader {
+    public final S stream;
+    private final short version;
+    private final BinaryStreamReader<S> reader;
+
+    public FastBinaryReader(final S stream, final short version) {
```

Comment:
Done. I think the compact binary equivalent in the other cr doesn't have the null check.